### PR TITLE
feat: collaboration mode API cleanup

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Project mode openapi specs change: Removing confusing nullable field. Making wording consistent

Depends on: https://github.com/Unleash/unleash/pull/3360

### CODE REVIEW

**Review by thomasheartman — CHANGES_REQUESTED** [review]
Thanks for cleaning up! 🙌🏼  

I think the description could change a little bit to be clearer, but the best would probably be to link to the docs. 

I've added a suggestion for the first instance, but I think the same suggestion should be added to the changes in `src/lib/openapi/spec/project-overview-schema.ts` and `src/lib/openapi/spec/project-schema.ts` too.

To be clear: I think the most future-proof solution is to simply link to the docs, but if you think project mode will only ever affect whether you can create change requests or not, then the longer version might be appropriate.

File: src/lib/openapi/spec/health-overview-schema.ts [hunk 1]
```diff
@@ -36,9 +36,8 @@ export const healthOverviewSchema = {
             type: 'string',
             enum: ['open', 'protected'],
             example: 'open',
-            nullable: true,
             description:
-                'A mode of the project affecting what actions are possible in this project. During a rollout of project modes this feature can be optional or `null`',
+                'A collaboration mode of the project affecting if non-project members can submit change requests',
```
[the remark this answers is not in the document]
Comment (kwasniew) [reply src/lib/openapi/spec/health-overview-schema.ts]: I like the first explicit variant. If we discover a different need for the project mode we can update later.

**Review by thomasheartman — APPROVED** [review]
Nice; I like it! But just so we're clear: is collaboration mode planned to control more than change requests?

**Comment by kwasniew** [comment]
@thomasheartman we don't know yet. For now it's only for change requests. We may or may not discover other use cases over time. I like to think of it as an option.

**Pull request merged**